tests/bluetooth/bsim_bt/bsim_test_gatt_caching/src/gatt_server_test.c

```diff
@@ -103,12 +122,7 @@ static void test_main_common(bool connect_eatt)
 	WAIT_FOR_FLAG(flag_is_connected);
 
 	if (connect_eatt) {
-		err = bt_eatt_connect(g_conn, CONFIG_BT_EATT_MAX);
-		if (err) {
-			FAIL("Failed to connect EATT channels (err %d)\n", err);
-
-			return;
-		}
+		WAIT_FOR_FLAG(flag_eatt_connect_success);
```

Comment:
It does seem like a WAIT_FOR_FLAG(flag_is_encrypted) before calling `bt_eatt_connect` would be better than calling `bt_eatt_connect` in the callback itself (fewer lines changed, more similar to your change in the other test, and the `security_changed` does not take `connect_eatt` into account

Reply:
indeed